### PR TITLE
Remove some CUDA nvcc suppression

### PR DESCRIPTION
**Pull request by cyyever**

### CODE REVIEW

File: cmake/public/cuda.cmake [hunk 1]
```diff
@@ -347,12 +347,7 @@ list(APPEND CUDA_NVCC_FLAGS ${NVCC_FLAGS_EXTRA})
 message(STATUS "Added CUDA NVCC flags for: ${NVCC_FLAGS_EXTRA}")
 
 # disable some nvcc diagnostic that appears in boost, glog, glags, opencv, etc.
-foreach(diag cc_clobber_ignored
-             set_but_not_used field_without_dll_interface
-             base_class_has_different_dll_interface
-             dll_interface_conflict_none_assumed
-             dll_interface_conflict_dllexport_assumed
-             bad_friend_decl)
+foreach(diag cc_clobber_ignored)
```
Comment (peterbell10) [inline cmake/public/cuda.cmake:350]: Have you checked the build logs if we run into any of these?

Comment (cyyever) [reply cmake/public/cuda.cmake:350]: No noticeable number of increased warnings. But I will re-check it locally.

**Review by peterbell10 — CHANGES_REQUESTED** [review]
Looks like there is lots of warning spam in the logs now, some on which is coming from third party libraries that we really don't want to fill the logs with.

**Pull request closed without merging**